ui/src/app/views/workspace-edit/component.ts

```diff
@@ -71,13 +84,34 @@ export class WorkspaceEditComponent implements OnInit {
       }
     }
   }
-
   navigateBack(): void {
     this.locationService.back();
   }
 
   resetWorkspaceCreation(): void {
     this.workspaceCreationError = false;
-    this.buttonClicked = false;
+    this.savingWorkspace = false;
+  }
+
+  updateWorkspace(): void {
+    if (!this.savingWorkspace) {
```

Comment:
Does savingWorkspace ever get cleared? I think it should be cleared on error at least, if not on success when we navigate away. (What happens if you save and then come back to the page -- does it reuse the same component & keep state?)

Similar questions for the other state fields.

Reply:
Yes, it gets cleared on error after the user hits `Keep Editing` on the modal that pops up. (Similarly for workspace creation error) If the user hits cancel, it navigates them to the previous page they were on.